droid-results/src/test/java/uk/gov/nationalarchives/droid/profile/ProfileContextLocatorTest.java

```diff
@@ -0,0 +1,190 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package uk.gov.nationalarchives.droid.profile;
+
+import org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream;
+import org.apache.commons.configuration2.PropertiesConfiguration;
+import org.apache.commons.compress.archivers.zip.ZipArchiveEntry;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.junit.Assert;
+import org.junit.Test;
+import uk.gov.nationalarchives.droid.core.interfaces.config.DroidGlobalConfig;
+import uk.gov.nationalarchives.droid.core.interfaces.signature.SignatureFileException;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+public class ProfileContextLocatorTest {
+
+    @Test
+    public void testOpenProfileInstanceManagerCreatesNewTemplateIfNewProfile() throws IOException, SignatureFileException, ConfigurationException {
+        String profileId = UUID.randomUUID().toString();
+        Path profileHome = Files.createDirectory(Files.createTempDirectory("profiles").resolve(profileId));
+        Path templatesDirectory = Files.createDirectory(profileHome.resolve("profile_templates"));
+        DroidGlobalConfig globalConfig = createTestConfig(profileHome);
+        Path signatureFilePath = profileHome.resolve("signature_files").resolve("testsigfile.xml");
+
+        ProfileInstanceLocator profileInstanceLocator = getProfileInstanceLocator(profileHome, profileId, globalConfig);
+
+        verify(profileInstanceLocator, times(2)).freezeDatabase(eq(profileId));
+        verify(profileInstanceLocator, times(2)).thawDatabase(eq(profileId));
+        verify(globalConfig).update(eq(Map.of("profile.binarySignatureLastUpdated.testsigfile.xml", Files.getLastModifiedTime(signatureFilePath).toMillis())));
+
+        List<String> templateFiles = Files.list(templatesDirectory)
+                .map(Path::getFileName)
+                .map(Path::toString)
+                .toList();
+        Assert.assertArrayEquals(templateFiles.toArray(), new String[]{"profile.1.template", "profile.template"});
```

Comment:
This seems to fail for me locally due to order being different. I thought `assertArrayEquals` is independent of order but looks like it is not. 🤔

Reply:
I've added in a sorted in the stream calls which should force it to be in the order that's there.